zenodo/modules/pages/views.py

```diff
@@ -51,39 +50,62 @@
 )
 def support():
     """Render contact form."""
-    form_class = contact_form_factory()
-    form = form_class()
+    form = ContactForm()
+    os, browser, device = user_agent_information()
 
-    if form.include_os_browser.data:
-        form.os.data = request.user_agent.platform
-        form.browser.data = request.user_agent.browser
-        form.version.data = request.user_agent.version
+    # If user is logged in add email to it's form.
     if current_user.is_authenticated:
         form.email.data = current_user.email
 
+    # Generation of message based upon maximum attachment size.
+    file_max_size = display_size(
+        current_app.config['PAGES_ATTACHMENT_MAX_SIZE'])
+
+    # Information added to form from config file.
+    form.issue_category.choices = [(row[0], row[0]) for row in
+                                   current_app.config['PAGES_ISSUE_CATEGORY']]
+    form.description.validators = [Length(
+        min=current_app.config['PAGES_DESCRIPTION_MIN_LENGTH'],
+        max=current_app.config['PAGES_DESCRIPTION_MAX_LENGTH'],
+    )]
+    form.attachments.description = "Optional. Max attachments size: " + \
+        file_max_size
+
     """If form is validated send email to the admin."""
     if form.validate_on_submit():
         """Dictionary storing data to be sent."""
         context = dict(
+            os=os,
+            browser=browser,
+            device=device,
             form=form,
             current_user=current_user,
         )
 
-        send_support_email(context)
-
-        flash(
-            _('Request sent successfully,'
-              'You should receive a confirmation email within 20 minutes - '
-              'if this does not happen you should retry or send us an email '
-              'directly to team@zenodo.org.',),
-            category='success'
-        )
-        return redirect('/')
+        if send_support_email(context) == "error":
+            flash(
+                _('File size exceeded, Please add url to file instead.'),
+                category='error',
+            )
+        else:
+            flash(
+                _('Request sent successfully, '
+                  'You should receive a confirmation email within 20 minutes -'
+                  ' if this does not happen you should retry or send us an '
+                  'email directly to team@zenodo.org.'),
+                category='success'
+            )
+            return redirect('/')
 
     content = current_app.config['PAGES_ISSUE_CATEGORY']
```

Comment:
You can move this line above, to keep everything related to the "view setup" together.